langchain/load/serializable.py

```diff
@@ -0,0 +1,93 @@
+from abc import ABC, abstractmethod
+from typing import TypedDict, Literal, Any
+
+from pydantic import BaseModel
+
+
+class BaseSerialized(TypedDict):
+    lc: int
+    type: str
+    id: list[str]
+
+
+class SerializedConstructor(BaseSerialized):
+    type: Literal["constructor"]
+    kwargs: dict[str, Any]
+
+
+class SerializedSecret(BaseSerialized):
+    type: Literal["secret"]
+
+
+class SerializedNotImplemented(BaseSerialized):
+    type: Literal["not_implemented"]
+
+
+class Serializable(BaseModel, ABC):
+    @property
+    def lc_namespace(self) -> list[str]:
+        """
+        Return the namespace of the langchain object.
+        eg. ["langchain", "llms", "openai"]
+        """
+        return self.__class__.__module__.split(".")
+
+    @property
+    def lc_secrets(self) -> dict[str, str]:
+        """
+        Return a map of constructor argument names to secret ids.
+        eg. {"openai_api_key": "OPENAI_API_KEY"}
+        """
+        return dict()
+
+    lc_kwargs: dict[str, Any] = dict()
```

Comment:
if we set in init is this needed?

Reply:
Probably not, will check and remove

Reply:
Actually yea it is otherwise pydantic errors when someone initialises a subclass with no args